scripts/tools/IndexSummary.php

```diff
@@ -68,15 +68,38 @@ protected function run(): Report
             $report = Report::createInfo($data['label']);
             $report->add(Report::createSuccess('Total in DB: ' . $data['totalInDb']));
             $report->add(Report::createSuccess('Total indexed "' . $data['index'] . '": ' . $totalIndexed));
-            $report->add(new Report($percentage < 100 ? Report::TYPE_ERROR : Report::TYPE_SUCCESS, 'Percentage indexed: ' . $percentage . '%'));
-            $report->add(new Report($missingIndex > 0 ? Report::TYPE_ERROR : Report::TYPE_SUCCESS, 'Missing items: ' . $missingIndex));
+            $report->add(
+                new Report(
+                    $this->getPercentageReportTypeBy($percentage, $missingIndex),
+                    'Percentage indexed: ' . $percentage . '%'
+                )
+            );
+            $report->add(
+                new Report(
+                    $missingIndex > 0 ? Report::TYPE_ERROR : Report::TYPE_SUCCESS,
+                    'Missing items: ' . $missingIndex
+                )
+            );
 
             $mainReport->add($report);
         }
 
         return $mainReport;
     }
 
+    private function getPercentageReportTypeBy(float $percentage, int $missingIndex): string
+    {
+        if ($missingIndex === 0) {
+            return Report::TYPE_WARNING;
+        }
```

Comment:
You probably need to check also if there is something in the DB to be indexed, no?

```suggestion
    private function getPercentageReportTypeBy(float $percentage, int $totalInDb, int $missingIndex): string
    {
        if ($totalInDb > 0 && $missingIndex === 0) {
            return Report::TYPE_WARNING;
        }
```

Reply:
upd. simplified as much as possible:
- if nothing in DB - warning (most probably everything is ok)
- if <100% indexed - error
- if 100% - success